realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -672,6 +692,22 @@ public Builder readOnly() {
             return this;
         }
 
+        /**
+         * Sets compactOnLaunch to determine if it should be compacted before returned to the user. It is passed the
+         * total file size (data + free space) and the total bytes used by data in the file.
+         *
+         * @param compactOnLaunch a callback called when opening a Realm for the first time during the life of a process
+         *                        to determine if it should be compacted before being returned to the user. It is passed
+         *                        the total file size (data + free space) and the total bytes used by data in the file.
+         */
+        public Builder compactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
```

Comment:
IMO we should also provide an easy to use configuration option with just a `compactOnLaunch()` override as well. Then we can provide our own default callback implementation `DefaultCompactOnLaunchCallback()`  with semantics provided by us.

I don't think that most users are in a better position that us to provide good heuristics around this, so we might as well make it really easy to use. 

Good starting heuristics could perhaps just be those used here:  https://news.realm.io/news/realm-objc-swift-2.6 although 100 MB feels a bit much on Android.